### PR TITLE
[Lean Core] Add deprecation notice to ImageStore

### PR DESCRIPTION
**Pull request by EvanBacon**
## Summary

- Related: #23313 
- ImageStore is **iOS only**. AFAIK there is no reason this functionality isn't available on Android.
- base64 is very inefficient with the React Native bridge
- Ideally the `FileSystem` solutions will integrate Turbo Modules to circumvent bridge issues by passing direct references to files.

## Changelog

* [General][added] - A deprecation notice with info about third-party solutions for getting a base64-encoded string.
* [General][fixed] - Missing warnings for unimplemented platform methods.

## Test Plan

* Run the console warnings in any React Native app to test the format.

### CODE REVIEW

**Comment by cpojer** [comment]
Thank you! 🙏

Could you change your implementation to use the `warnOnce` module that's already in the repo? We shouldn't spam the yellowbox multiple times for each module.

**Comment by TheSavior** [comment]
Should these warnings be added in react-native-implementation.js? That is where we've put the other warnings.

File: Libraries/react-native/react-native-implementation.js [hunk 1]
```diff
@@ -50,6 +50,13 @@ module.exports = {
     return require('ImageEditor');
   },
   get ImageStore() {
+    warnOnce(
+      'imagestore-deprecation',
+      'ImageStore is deprecated and will be removed in a future release. ' +
+        'To get a base64-encoded string from a local image use either of the following third-party libraries:' +
+        '* expo-file-system: `readAsStringAsync(filepath, \'base64\')`' +
```
[the remark this answers is not in the document]
Comment (EvanBacon) [reply Libraries/react-native/react-native-implementation.js]: It'd be cool if this made a `suggestion` block

**Comment by EvanBacon** [comment]
Could the failure be unrelated?
```
 FAIL  RNTester/e2e/__tests__/Switch-test.js (38.853s)
  Switch
    Switches can be set to true or false
      ✓ Switch that starts off should switch (732ms)
      ✕ Switch that starts on should switch (652ms)
    Switches can be disabled
      ✓ disabled switch should not toggle (300ms)

  ● Switch › Switches can be set to true or false › Switch that starts on should switch
```

**Pull request closed without merging**